analyticsdataserver/clients.py

```diff
@@ -22,16 +22,31 @@ class CourseBlocksApiClient(EdxRestApiClient):
     Currently, this client is only used for a local-only developer script (generate_fake_course_data).
     """
 
-    def __init__(self, url, access_token, timeout):
-        super().__init__(url, oauth_access_token=access_token, timeout=timeout)
-
     def all_videos(self, course_id):
         try:
             logger.debug('Retrieving course video blocks for course_id: %s', course_id)
-            response = self.blocks.get(course_id=course_id, all_blocks=True, depth='all', block_types_filter='video')
+
+            try:
+                api_base_url = settings.LMS_BASE_URL + 'api/courses/v1/'
+            except AttributeError:
+                logger.warning("LMS_BASE_URL is not configured! Cannot get video ids.")
+                return None
+            logger.info("Assuming the Course Blocks API is hosted at: %s", api_base_url)
+
+            blocks_kwargs = {
+                'course_id': course_id,
+                'all_blocks': True,
+                'depth': 'all',
+                'block_types_filter': 'video'
+            }
+            response = self.get(f'{api_base_url}blocks/', params=blocks_kwargs)
+            response.raise_for_status()
+            data = response.json()
             logger.info("Successfully authenticated with the Course Blocks API.")
-        except HttpClientError as e:
-            if e.response.status_code == 401:
+        except RequestException as e:
```

Comment:
raise_for_status raises [HTTPError](https://2.python-requests.org/en/master/api/#requests.HTTPError)  why do you prefer handling the RequestException one?